### PR TITLE
circuit-types: Refactor application level types over new proof system

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR finishes refactoring all types in the `circuit-types` crate over the new proof system. This largely involves deleting code and fixing imports where traits have changed.

### Testing
- Unit tests pass

### CODE REVIEW

File: circuit-types/src/lib.rs [hunk 1]
```diff
@@ -37,13 +38,25 @@ use wallet::{Wallet, WalletShare};
 // | Constants |
 // -------------
 
+/// The zero value in the Scalar field we work over
+pub const SCALAR_ZERO: ScalarField = ScalarField::new(BigInt::new([0, 0, 0, 0]));
+/// The one value in the Scalar field we work over
+pub const SCALAR_ONE: ScalarField = ScalarField::new(BigInt::new([1, 0, 0, 0]));
```
Comment (akirillo) [inline circuit-types/src/lib.rs:44]: Confirming this is correct even in Montgomery form?

File: circuit-types/src/lib.rs [hunk 2]
```diff
@@ -318,8 +259,11 @@ pub mod native_helpers {
     {
         // Sample new wallet blinders from the `blinder` CSPRNG
         // See the comments in `valid_reblind.rs` for an explanation of the two CSPRNGs
-        let mut blinder_samples =
-            evaluate_hash_chain(private_secret_shares.blinder, 2 /* length */);
+        let mut blinder_samples = evaluate_hash_chain(
+            private_secret_shares.blinder,
+            2, /* length
+                */
```
Comment (akirillo) [inline circuit-types/src/lib.rs:265]: was this `cargo fmt`?

File: circuit-types/Cargo.toml [hunk 3]
```diff
@@ -10,6 +10,7 @@ edition = "2021"
 ark-ff = "0.4"
 ark-ec = "0.4"
 ark-mpc = { workspace = true }
+ed25519-dalek = "1.0.1"
```
Comment (akirillo) [inline circuit-types/Cargo.toml:13]: I think we can remove this?

**Review by akirillo — APPROVED** [review]
couple small questions

File: circuit-types/src/lib.rs (hunk 1 above)
Comment (joeykraut) [reply circuit-types/src/lib.rs:44]: The `BigInt` constructor is not concerned with montgomery form, only the `Fp` (which is a `MontgomeryBackend`) will make such changes to the underlying bigint

File: circuit-types/Cargo.toml (hunk 3 above)
Comment (joeykraut) [reply circuit-types/Cargo.toml:13]: No, we use it for keys for the moment

File: circuit-types/src/lib.rs (hunk 2 above)
Comment (joeykraut) [reply circuit-types/src/lib.rs:265]: it was, i fixed it

**Pull request merged**